### PR TITLE
Fix deployment delete

### LINKED ISSUE #6289: 🐛 Bug Report: Deleting deployment does not work for VCS deployments
### 👟 Reproduction steps

1. Setup S3 Storage (Digital Ocean Spaces)
2. Deploy any function
3. Go to that function and delete any deployments
4. Deployment does not get deleted

Logs from worker-deletes:
Screenshot 2023-09-21 at 03 17 20@2x

### 👍 Expected behavior

Delete that specific deployment from storage

### 👎 Actual Behavior

It actually does not delete it

### 🎲 Appwrite version

Version 1.4.x

### 💻 Operating system

Linux

### 🧱 Your Environment

Version 1.4.3 Self hosted

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by stnguyen90**
## What does this PR do?

Prior to this, deleting a VCS deployment would fail because the deployment path for VCS deployments are empty. Because an error gets thrown, the delete function would stop and not delete the build files.

This change updates the worker to only delete the storage files if the path is set.

Fixes #6289

## Test Plan

Manually deleted a deployment:

Screen Shot 2023-09-20 at 7 29 53 PM

Manually deleted a function

Screen Shot 2023-09-20 at 7 30 14 PM

## Related PRs and Issues

- #6289

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [x] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

**Review by Meldiron — APPROVED** [review]

**Review by TorstenDittmann — APPROVED** [review]

File: app/workers/deletes.php [hunk 1]
```diff
@@ -679,13 +724,8 @@ protected function deleteDeployment(Document $document, Document $project): void
...
          * Delete deployment files
          */
-        Console::info("Deleting deployment files for deployment " . $deploymentId);
         $storageFunctions = $this->getFunctionsDevice($projectId);
-        if ($storageFunctions->delete($document->getAttribute('path', ''), true)) {
-            Console::success('Deleted deployment files: ' . $document->getAttribute('path', ''));
-        } else {
-            Console::error('Failed to delete deployment files: ' . $document->getAttribute('path', ''));
-        }
+        $this->deleteDeploymentFiles($storageFunctions, $document);
```
Comment (christyjacob4) [inline app/workers/deletes.php:728]:
```suggestion
        $this->deleteDeploymentFiles($deviceFunctions, $document);
```

File: app/workers/deletes.php [hunk 2]
```diff
@@ -695,11 +735,7 @@ protected function deleteDeployment(Document $document, Document $project): void
         $this->deleteByGroup('builds', [
             Query::equal('deploymentInternalId', [$deploymentInternalId])
         ], $dbForProject, function (Document $document) use ($storageBuilds) {
-            if ($storageBuilds->delete($document->getAttribute('path', ''), true)) {
-                Console::success('Deleted build files: ' . $document->getAttribute('path', ''));
-            } else {
-                Console::error('Failed to delete build files: ' . $document->getAttribute('path', ''));
-            }
+            $this->deleteBuildFiles($storageBuilds, $document);
```
Comment (christyjacob4) [inline app/workers/deletes.php:738]:
```suggestion
            $this->deleteBuildFiles($deviceBuilds, $document);
```

File: app/workers/deletes.php [hunk 3]
```diff
@@ -626,11 +627,7 @@ protected function deleteFunction(Document $document, Document $project): void
             Query::equal('resourceInternalId', [$functionInternalId])
         ], $dbForProject, function (Document $document) use ($storageFunctions, &$deploymentInternalIds) {
             $deploymentInternalIds[] = $document->getInternalId();
-            if ($storageFunctions->delete($document->getAttribute('path', ''), true)) {
-                Console::success('Deleted deployment files: ' . $document->getAttribute('path', ''));
-            } else {
-                Console::error('Failed to delete deployment files: ' . $document->getAttribute('path', ''));
-            }
+            $this->deleteDeploymentFiles($storageFunctions, $document);
```
Comment (christyjacob4) [inline app/workers/deletes.php:630]:
```suggestion
            $this->deleteDeploymentFiles($deviceFunctions, $document);
```

File: app/workers/deletes.php [hunk 4]
```diff
@@ -642,11 +639,7 @@ protected function deleteFunction(Document $document, Document $project): void
             $this->deleteByGroup('builds', [
                 Query::equal('deploymentInternalId', [$deploymentInternalId])
             ], $dbForProject, function (Document $document) use ($storageBuilds) {
-                if ($storageBuilds->delete($document->getAttribute('path', ''), true)) {
-                    Console::success('Deleted build files: ' . $document->getAttribute('path', ''));
-                } else {
-                    Console::error('Failed to delete build files: ' . $document->getAttribute('path', ''));
-                }
+                $this->deleteBuildFiles($storageBuilds, $document);
```
Comment (christyjacob4) [inline app/workers/deletes.php:642]:
```suggestion
                $this->deleteBuildFiles($deviceBuilds, $document);
```

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Looks Good 👌  Please address the comments

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**